backend/dataall/modules/notebooks/cdk/notebook_policy.py

```diff
@@ -25,125 +43,169 @@ def get_statements(self, group_permissions, **kwargs):
             iam.PolicyStatement(
-                actions=['sagemaker:InvokeEndpoint', 'sagemaker:InvokeEndpointAsync'],
+                # sid="SageMakerApps",
+                effect=iam.Effect.ALLOW,
+                actions=[
+                    'sagemaker:CreateApp',
+                    'sagemaker:DeleteApp'
+                ],
+                resources=[f'arn:aws:sagemaker:{self.region}:{self.account}:app/*/*']
+            ),
+            iam.PolicyStatement(
+                # sid="SageMakerCreatePresignedDomainUrl",
+                effect=iam.Effect.ALLOW,
+                actions=['sagemaker:CreatePresignedDomainUrl'],
+                resources=[f'arn:aws:sagemaker:{self.region}:{self.account}:user-profile/*/*'],
+                conditions={
+                    'StringEquals': {
+                        f'sagemaker:ResourceTag/{self.tag_key}': [self.tag_value]
+                    },
+                },
+            ),
+            iam.PolicyStatement(
+                # sid="SageMakerManageGenericResources",
+                effect=iam.Effect.ALLOW,
+                actions=[
+                    'sagemaker:Delete*',
+                    'sagemaker:Update*',
+                    'sagemaker:Start*',
+                    'sagemaker:Stop*',
+                    'sagemaker:InvokeEndpoint',
+                    'sagemaker:InvokeEndpointAsync'
+                ],
                 resources=[
-                    f'arn:aws:sagemaker:{self.region}:{self.account}:endpoint/*'
+                    f'arn:aws:sagemaker:{self.region}:{self.account}:*/{self.resource_prefix}*',
+                    f'arn:aws:sagemaker:{self.region}:{self.account}:*/{self.resource_prefix}*/*',
                 ],
                 conditions={
                     'StringEquals': {
-                        f'aws:ResourceTag/{self.tag_key}': [self.tag_key]
-                    }
+                        f'aws:ResourceTag/{self.tag_key}': [self.tag_value],
+                    },
                 },
             ),
+            # Logging and support permissions
             iam.PolicyStatement(
+                # sid="SageMakerLogging",
+                effect=iam.Effect.ALLOW,
                 actions=[
                     'logs:CreateLogGroup',
                     'logs:CreateLogStream',
-                    'logs:PutLogEvents'],
+                    'logs:PutLogEvents'
+                ],
                 resources=[
                     f'arn:aws:logs:{self.region}:{self.account}:log-group:/aws/sagemaker/*',
                     f'arn:aws:logs:{self.region}:{self.account}:log-group:/aws/sagemaker/*:log-stream:*',
                 ]
             ),
             iam.PolicyStatement(
+                # sid="SageMakerSupport",
+                effect=iam.Effect.ALLOW,
                 actions=[
                     'ecr:GetAuthorizationToken',
                     'ecr:BatchCheckLayerAvailability',
                     'ecr:GetDownloadUrlForLayer',
```

Comment:
Part of the #TODO is to split this policy into notebook_policy and mlstudio_policy